src/js/controllers/mapController.ts

```diff
@@ -535,6 +591,16 @@ export class MapController {
     }
   }
 
+  updateMeasureWidgetOnClick(
+    selectedWidget: DistanceMeasurement2D | AreaMeasurement2D
+  ): void {
+    const mapviewOnClick = this._mapview?.on('click', event => {
```

Comment:
let's think about throwing `once` here!

Reply:
@vaidotasp @SwampGuzzler Thanks so much for the feedback! I tried leveraging `once`, as well as a few other methods in ESRI's `watchUtils`,  but it looks like it wasn't able to fire `.newMeasurement()`